kore/src/Kore/Builtin/Builtin.hs

```diff
@@ -483,3 +503,37 @@ makeDomainValuePattern ::
 makeDomainValuePattern sort stringLiteral =
     Pattern.fromTermLike $
         makeDomainValueTerm sort stringLiteral
+
+-- | Unification of @eq@ symbols
+unifyEq ::
+    forall unifier unifyData.
+    MonadUnify unifier =>
+    (unifyData -> EqTerm (TermLike RewritingVariableName)) ->
+    (unifyData -> InternalBool) ->
```

Comment:
The three `Unify..Eq` types are basically the same, would it make sense to just have a single `UnifyEq` type in these cases?